### PR TITLE
Adds a MacOS test workflow

### PR DESCRIPTION
**Pull request by kaczmarczyck**
`bazel test` is run on Linux and MacOS with this PR.

### CODE REVIEW

File: .github/workflows/bazel_test_mac.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Copyright 2020 Google LLC
```
Comment (jmichelp) [inline .github/workflows/bazel_test_mac.yml:1]: 2021?

Comment (kaczmarczyck) [reply .github/workflows/bazel_test_mac.yml]: Done

**Comment by kaczmarczyck** [comment]
The MacOS tests are currently queued for quite some time before running.

**Comment by jmichelp** [comment]
It's probably because they don't have a lot of Big Sur machines at the moment (it's a preview).
Try to change it for `macos-10.15` instead.

**Comment by kaczmarczyck** [comment]
The older version was faster, so I'll take that.

**Review by jmichelp — APPROVED** [review]

**Pull request merged**